src/mpid/ch4/shm/posix/eager/iqueue/iqueue_send.h

```diff
@@ -19,116 +19,75 @@ MPL_STATIC_INLINE_PREFIX size_t MPIDI_POSIX_eager_buf_limit(void)
     return MPIR_CVAR_CH4_SHM_POSIX_IQUEUE_CELL_SIZE;
 }
 
-/* This function attempts to send the next chunk of a message via the queue. If no cells are
- * available, this function will return and the caller is expected to queue the message for later
- * and retry.
- *
- * grank   - The global rank (the rank in MPI_COMM_WORLD) of the receiving process.
- * msg_hdr - The header of the message to be sent. This can be NULL if there is no header to be sent
- *           (such as if the header was sent in a previous chunk).
- * iov     - The array of iovec entries to be sent.
- * iov_num - The number of entries in the iovec array.
+/* This function attempts to allocate a cell and return the address and size of the payload portion
+ * of the cell.
  */
-MPL_STATIC_INLINE_PREFIX int
-MPIDI_POSIX_eager_send(int grank,
-                       MPIDI_POSIX_am_header_t ** msg_hdr, struct iovec **iov, size_t * iov_num)
+MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_eager_get_buf(void **eager_buf, size_t * eager_buf_sz)
```

Comment:
This commit looks like it might address some existing issues in the shm layer. Does it depend on the new framework, or could it be separated out?

Reply:
I think this "refactoring POSIX EAGER" does not rely on the new framework. Let me try make it a separate PR.

Reply:
After a closer look, this one commit won't work standalone because it was breaking the existing pipeline implementation inside POSIX.